File: VirtualAid/backend/aspnet-core/src/VirtaulAid.Domain/Companies/Company.cs
using Microsoft.AspNetCore.Http;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using VirtaulAid.Carts;
using VirtaulAid.Courses;
using VirtaulAid.Purchases;
using VirtaulAid.Users;
using Volo.Abp.Domain.Entities.Auditing;

namespace VirtaulAid.Companies
{
    public class Company : FullAuditedAggregateRoot<Guid>
    {
        public string CompanyName { get; set; }
        public string Email { get; set; }
        public string ContactNumber { get; set; }
        public string? Address1 { get; set; }
        public string? Address2 { get; set; }
        public string? Address3 { get; set; }
        public string? Country { get; set; }
        public string? State { get; set; }
        public string? City { get; set; }
        public string? Postalcode { get; set; }
        public bool? IsVerified { get; set; }
        public bool IsLocked { get; set; } = false;
        public string? ProfileImage { get; set; }
        public string? BannerImage { get; set; }
        public string? Slogan { get; set; }
        public string? Bio { get; set; }
        public int? NoOfEmployees { get; set; }
        public bool? PublishData { get; set; } = false;
        public virtual ICollection<Cart>? Carts { get; set; }
        public virtual ICollection<PurchaseDetail>? PurchaseDetails { get; set; }
        public virtual ICollection<UserDetail>? Employees { get; set; }
        public virtual ICollection<CourseSubscriptionMapping>? CourseSubscriptionMappings { get; set; }
        public virtual ICollection<LoggedInUser>? LoggedInUsers { get; set; }

    }
}

File: VirtualAid/frontend-nx/apps/individual/types/company.ts
export type CompanyEmployeeType = {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  contactNumber: string;
  address1: string;
  address2: string;
  address3: string;
  country: string;
  state: string;
  city: string;
  postalcode: string;
  latitude: number;
  longitude: number;
  currentCompanyId: string;
  totalCourses: number;
  certificateExpirationDate: string;
};

export type CompanyInfoType = {
  id: string;
  creationTime: string;
  creatorId: string;
  lastModificationTime: string;
  lastModifierId: string;
  companyName: string;
  email: string;
  contactNumber: string;
  designation: string;
  address1: string;
  address2: string;
  address3: string;
  bio: string;
  country: string;
  state: string;
  city: string;
  postalcode: string;
  isVerified: boolean;
  isLocked: boolean;
  profileImage: string;
  bannerImage:string;
  slogan: string;
  noOfEmployees: number;
};

export type CourseEnrolledEmployeeType = CompanyEmployeeType & {
  enrolledDate: string;
  progress: number;
  isPersonalCourse: boolean;
};

export type TerminatedEmployeeType = {
  id: string;
  companyId: string;
  userId: string;
  terminationDate: string;
  employeeName: string;
  employeeEmail: string;
};
